lib/fizzy/value.hpp

```diff
@@ -0,0 +1,40 @@
+// Fizzy: A fast WebAssembly interpreter
+// Copyright 2019-2020 The Fizzy Authors.
+// SPDX-License-Identifier: Apache-2.0
+
+#pragma once
+
+#include <cstdint>
+
+namespace fizzy
+{
+union Value
+{
+    uint64_t i64;
+    float f32;
+
+    Value() = default;
+
+    constexpr Value(uint64_t v) noexcept : i64{v} {}
+    constexpr Value(uint32_t v) noexcept : i64{v} {}
+    constexpr Value(int64_t v) noexcept : i64{static_cast<uint64_t>(v)} {}
+    constexpr Value(int32_t v) noexcept : i64{static_cast<uint32_t>(v)} {}
+
+    constexpr Value(float v) noexcept : f32{v} {}
+
+    constexpr operator uint64_t() const noexcept { return i64; }
+
+    /// Get the value as the given type. Handy in templates.
+    template <typename T>
+    constexpr T as() const noexcept
```

Comment:
Can this be left without default implementation and then have specializations for `uint32_t` and `uint64_t` similar to `float`?

Reply:
Done in #423.